pkg/utils/overriding/merging.go

```diff
@@ -84,45 +91,49 @@ func mergeKeyedDevWorkspaceTemplateSpec(
 		return nil, err
 	}
 
-	result := keyedDevWorkspaceTemplateSpecContent{}
+	result := workspaces.DevWorkspaceTemplateSpecContent{}
 
 	preStartCommands := sets.String{}
 	postStartCommands := sets.String{}
 	preStopCommands := sets.String{}
 	postStopCommands := sets.String{}
 
-	for _, content := range allContents {
-		if content.Commands != nil && len(content.Commands) > 0 {
-			if result.Commands == nil {
-				result.Commands = []keyedCommand{}
-			}
-			for _, command := range content.Commands {
-				result.Commands = append(result.Commands, command)
-			}
+	topLevelListsNames := result.GetToplevelLists()
+	resultValue := reflect.ValueOf(&result).Elem()
+	componentType := reflect.TypeOf(workspaces.Component{})
```

Comment:
There has to be a better way to do this than by reflection, the loops below are *so* complicated. I think it'd even be simpler to have copy-pasted code for each case here.

Reply:
I did an attempt at simplifying it in commit: https://github.com/devfile/api/pull/122/commits/88d00ae9e2509417ac4f0c9118b5f2c52488e00b

Reply:
LGTM, thanks.